### ISSUE TITLE
Goto def to methods called with overloaded operator ->

### ISSUE THREAD

**Issue by Roman-Skabin** [slynch8/10x#1219]
It's more a question than an issue.

For example, you have a class or a struct with a member functions:
```C++
struct SomeStruct
{
    void SomeMethod() {}
};
```
And you have another struct like smart pointer or something which overloads operator->. So when you have somethiing like
```C++
void foo()
{
    // ...
    SmartPointer<SomeStruct> someStruct;
    // ...
    someStruct->SomeMethod();
    // ...
}
```
in the code, `SomeMethod` highlights correctly, but goto def does not work for it.

**Comment by Roman-Skabin** [comment]
Yeah, so the question itself 😃. Is it a bug or you just do not support goto def for that case yet?

**Comment by slynch8** [comment]
That should work, and works in my tests. Maybe it's not parsing the -> operator in your SmartPointer class for some reason. Can you give a simple example that shows the problem?

**Comment by slynch8** [comment]
Thanks for the repro. Worked perfectly. Fixed (0.34.32)

**Comment by Roman-Skabin** [comment]
Still does not work. Even for such a simple example:
```C++
#include <memory>

struct A
{
    void foo() {}
};

void bar()
{
    std::unique_ptr<A> a;
    a->foo();
}
```
I press F12 and it does not jump anywhere. The cursor just remains at the line with `a->foo();`. Auto complete does not work as well.

If it is important I do not create .10x_workspace file. I open .sln file generated with cmake 3.14 for visual studio 2019.
\
\
\-------------------
Version: 0.34.42

**Comment by slynch8** [comment]
fixed (0.35.0)
